source/common/http/conn_manager_impl.cc

```diff
@@ -1120,6 +1186,40 @@ void ConnectionManagerImpl::ActiveStreamEncoderFilter::
 
 void ConnectionManagerImpl::ActiveStreamEncoderFilter::continueEncoding() { commonContinue(); }
 
+void ConnectionManagerImpl::ActiveStreamEncoderFilter::responseDataTooLarge() {
+  if (parent_.encoder_filters_streaming_) {
+    onEncoderFilterAboveWriteBufferHighWatermark();
+  } else {
+    // If headers have not been sent to the user, send a 500.
+    if (!headers_continued_) {
+      // Make sure we won't end up with nested watermark calls from the body buffer.
+      parent_.encoder_filters_streaming_ = true;
+      stopped_ = false;
+      HeaderMapPtr response_headers{new HeaderMapImpl{
```

Comment:
This is basically sendLocalReply. Is there any way to use that here?

Reply:
Is it kosher to have an encoder filter use parent.decoder_filters_.begin()?  It shouldn't be deleted until onDestroy but feels super hacky to me.

Reply:
Sorry not following. Why is that needed?

Reply:
Oh I see because it takes a StreamDecoderFilterCallbacks. Alright not a big deal. One option would be to have a version that takes lambdas for encoding headers and data, then use that version in both your place and the existing utility function. (There might be other options also). Just use your discretion here.